plugins/contract-sizes.ts

```diff
@@ -0,0 +1,575 @@
+import {Table} from 'console-table-printer'
+import {extendConfig, task, types} from 'hardhat/config'
+import {
+    HardhatConfig,
+    HardhatRuntimeEnvironment,
+    SolcConfig
+} from 'hardhat/types'
+import chalk from 'chalk'
+
+import fs from 'fs'
+import path from 'path'
+
+const MAX_CONTRACT_SIZE = 24576 // applied to color contract size as green / yellow / red
+
+/* eslint-disable no-lone-blocks */
+
+extendConfig((config: HardhatConfig) => {
+    for (const compiler of config.solidity.compilers) {
+        addOutputSelections(compiler, [
+            'evm.bytecode.object',
+            'evm.bytecode.sourceMap',
+            'evm.deployedBytecode.object',
+            'evm.deployedBytecode.sourceMap'
+        ])
+    }
+})
+
+task(
+    'contract-sizes',
+    'Prints size of contracts, including contribution of source files into bytecode of a contract'
+)
+    .addFlag(
+        'details',
+        'Print contribution of each source files into bytecode of a contract'
+    )
+    .addFlag(
+        'diff',
+        'Print size difference with the previous run with this flag'
+    )
+    .addOptionalParam(
+        'size',
+        'Filter by contract size (20 000 by default)',
+        0,
+        types.int
+    )
+    .addOptionalVariadicPositionalParam(
+        'contracts',
+        'Contracts to be printed, names or FQNs'
+    )
+    .setAction(async ({details: verbose, diff, size, contracts}, hre) => {
+        const c = (contracts ?? []) as string[]
+        const mappings = await extractBytecodeMappings(hre, c)
+        if (mappings.length === 0) {
+            // eslint-disable-next-line no-console
+            console.log('No contracts found')
```

Comment:
suggestion: use a logger over sop.

Bit more of a paint to setup, however it grants the user greater choice in what is done with the output

Reply:
afaik hardhat uses console, other hardhat plugins are using console ... why should this hardhat plugin be different?